packages/core/admin/admin/src/utils/getFetchClient.ts

```diff
@@ -111,20 +91,133 @@ type FetchClient = {
  * };
  * ```
  */
-const getFetchClient = (defaultOptions: AxiosRequestConfig = {}): FetchClient => {
-  instance.defaults.baseURL = window.strapi.backendURL;
-  return {
-    get: (url, config) =>
-      instance.get(normalizeUrl(url), {
-        ...defaultOptions,
-        ...config,
-      }),
-    put: (url, data, config) =>
-      instance.put(normalizeUrl(url), data, { ...defaultOptions, ...config }),
-    post: (url, data, config) =>
-      instance.post(normalizeUrl(url), data, { ...defaultOptions, ...config }),
-    del: (url, config) => instance.delete(normalizeUrl(url), { ...defaultOptions, ...config }),
+const getFetchClient = (defaultOptions: FetchConfig = {}): FetchClient => {
+  const { options } = defaultOptions;
+  const headers = new Headers({
+    Accept: 'application/json',
+    'Content-Type': 'application/json',
+    Authorization: `Bearer ${getToken()}`,
+  });
+
+  // Add a response interceptor to return the response
+  const responseInterceptor = async <TData = unknown>(
+    response: Response
+  ): Promise<FetchResponse<TData>> => {
+    const result = await response.json();
+
+    return {
+      data: result,
+    };
+  };
+
+  const paramsSerializer = <Param = unknown>(url: string, params?: Param) => {
+    if (params) {
+      const serializedParams = qs.stringify(params, { encode: false });
+      return `${url}?${serializedParams}`;
+    }
+    return url;
+  };
+
+  const addBaseUrl = (url: FetchURL, clientBaseURL?: string) => {
+    if (clientBaseURL) {
+      return `${clientBaseURL}${url}`;
+    }
+    if (options?.baseURL) {
+      return `${options?.baseURL}${url}`;
+    }
+    const baseURL = window.strapi.backendURL;
```

Comment:
would it make sense to have `window.strapi.backendURL` as the default value for the second argument?